Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS.py

```diff
@@ -0,0 +1,1375 @@
+        url_suffix = f"/passwordvault/api/Accounts/{account_id}/Password/Update"
+        body = {
+            "NewCredentials": new_credentials,
+        }
+        self._http_request("POST", url_suffix, json_data=body, resp_type='text')
+
+    def verify_credentials(self,
+                           account_id: str,
+                           ):
+        """
+        This function uses the V1 CyberArk PAS api, currently there is no matching function in V2
+        """
+        url_suffix = f"/PasswordVault/WebServices/PIMServices.svc/Accounts/{account_id}/VerifyCredentials"
+
+        self._http_request("POST", url_suffix, resp_type='text')
+
+    def reconcile_credentials(self,
+                              account_id: str,
+                              ):
+        url_suffix = f"/PasswordVault/API/Accounts/{account_id}/Reconcile"
+
+        self._http_request("POST", url_suffix, resp_type='text')
+
+    def update_account(self,
+                       account_id: str,
+                       account_name: str,
+                       address: str,
+                       username: str,
+                       platform_id: str,
+                       ):
+        url_suffix = f"/PasswordVault/api/Accounts/{account_id}"
+
+        arguments = {"name": account_name,
+                     "address": address,
+                     "userName": username,
+                     "platformId": platform_id}
+        body = []
+        for key, value in arguments.items():
+            if value:
+                body.append(
+                    {
+                        "op": "replace",
+                        "path": f"/{key}",
+                        "value": f"{value}"
+                    }
+                )
+
+        return self._http_request("PATCH", url_suffix, data=str(body))
+
+    def get_security_events(self,
+                            next_run: str
+                            ):
+        url_suffix = "/PasswordVault/API/pta/API/Events/"
+        self._headers["lastUpdatedEventDate"] = next_run
+        return self._http_request("GET", url_suffix)
+
+
+def test_module(
+        client: Client,
+) -> str:
+    """
+    If a client was made then an accesses token was successfully reached,
+    therefor, the username and password are valid and a connection was made
+    checks that the fetch command works as well, using the client function -
+    :param client: the client object with an access token
+    :return: ok if got a valid accesses token
+    """
+    start, _ = parse_date_range("7 days")
+    start_time_timestamp = str(date_to_timestamp(start))
+    security_events = client.get_security_events(start_time_timestamp)
+    # if there were security events in the last week
+    if security_events:
+        event = security_events[0]
+        if event.get("id") and event.get("type") and event.get("score"):
+            return "ok"
+        else:
+            raise Exception("Security events are not in the right format")
```

Comment:
more elegant to have only one return statement. WDYT?
```suggestion
        if not event.get("id") or not event.get("type") or no event.get("score"):
            raise Exception("Security events from CyberArk PAS are missing mandatory fields.")
```

Reply:
looks much better